### DISCUSSION TITLE
Automatic resize after hide/show polybar in bspwm

### DISCUSSION THREAD

**Discussion by developer1105** [polybar/polybar#2701] [Ideas]
When hiding/showing polybar in bspwm, it is necessary to issue a command after polybar-toggle, so that the window gets resized. 

Every option in polybar's config file is set to bspwm instead of i3. Wouldn't it be interesting if polybar itself issued the command "bspc config top_padding" right after toggling the visibility of the bar?

```
# sxhkd config excerpt

super + @b
    polybar-toggle; bspc config top_padding {0, 26}
```

**Comment by patrick96** [comment]
Two things:

Polybar already kind of does this. When polybar starts, it tells the WM how much space it needs at the edge of the screen and bspwm (and many other WMs) add a gap for the bar (`bspc config top_padding 26` isn't actually needed, this happens automatically). However, when the bar is hidden, it doesn't tell the WM that it no longer requires space. That's not really intentional, and I created a fix for that #2702.

Secondly, it seems bspwm does not reset the gap when the bar window is removed. So even with my changes #2702, bspwm does not remove the gap despite polybar no longer requesting a gap.
I asked about this in https://github.com/baskerville/bspwm/issues/1382.

**Reply by developer1105** [reply]
Thanks, you guys are great!
